SonarQube.TeamBuild.PreProcessor/SonarQubeServerFactory.cs

```diff
@@ -0,0 +1,35 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="SonarQubeServerFactory.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using SonarQube.Common;
+using System;
+using System.Net;
+
+namespace SonarQube.TeamBuild.PreProcessor
+{
+    internal class SonarQubeServerFactory : ISonarQubeServerFactory
+    {
+        public ISonarQubeServer Create(ProcessedArgs args)
+        {
+            if (args == null)
+            {
+                throw new ArgumentNullException("args");
+            }
+
+            return new SonarWebService(GetDownloader(args), args.GetSetting(SonarProperties.HostUrl));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) In method 'SonarQubeServerFactory.Create(ProcessedArgs)', call System.IDisposable.Dispose on object 'new SonarWebService(SonarQubeServerFactory.GetDownloader(args), args.GetSetting("sonar.host.url"))' before all references to it are out of scope. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=fxcop%3ADisposeObjectsBeforeLosingScope)